probe-rs/src/bin/probe-rs/cmd/debug.rs

```diff
@@ -768,58 +768,62 @@ impl DebugCli {
         });
 
         cli.add_command(Command {
-            name: "dump",
-            help_text: "Store a dump of the current CPU state",
-
-            function: |cli_data, _args| {
-                // dump all relevant data, stack and regs for now..
-                //
-                // stack beginning -> assume beginning to be hardcoded
-
-                let stack_top: u32 = 0x2000_0000 + 0x4000;
-
-                let stack_bot: u32 = cli_data.core.read_core_reg(cli_data.core.stack_pointer())?;
-                let pc: u32 = cli_data
-                    .core
-                    .read_core_reg(cli_data.core.program_counter())?;
-
-                let mut stack = vec![0u8; (stack_top - stack_bot) as usize];
-
-                cli_data.core.read(stack_bot.into(), &mut stack[..])?;
-
-                let mut dump = Dump::new(stack_bot, stack);
-
-                for i in 0..12 {
-                    dump.regs[i as usize] =
-                        cli_data.core.read_core_reg(Into::<RegisterId>::into(i))?;
-                }
-
-                dump.regs[13] = stack_bot;
-                dump.regs[14] = cli_data
-                    .core
-                    .read_core_reg(cli_data.core.return_address())?;
-                dump.regs[15] = pc;
-
-                let serialized = ron::ser::to_string(&dump).expect("Failed to serialize dump");
+            name: "reset",
 
-                let mut dump_file = File::create("dump.txt").expect("Failed to create file");
+            help_text: "Reset the CPU",
 
-                dump_file
-                    .write_all(serialized.as_bytes())
-                    .expect("Failed to write dump file");
+            function: |cli_data, _args| {
+                cli_data.core.halt(Duration::from_millis(100))?;
+                cli_data.core.reset_and_halt(Duration::from_millis(100))?;
 
                 Ok(CliState::Continue)
             },
         });
 
         cli.add_command(Command {
-            name: "reset",
+            name: "dump",
+            help_text: "Dump the core memory & registers",
 
-            help_text: "Reset the CPU",
+            function: |cli_data, args| {
+                let location = Path::new("./coredump");
```

Comment:
Can we make this an optional argument, and only use "./coredump" as default? It would be nice to not be tied to a hardcoded value :-)